src/openforms/contrib/kadaster/api/views.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import re
+
+from django.conf import settings
+from django.http import HttpResponseBadRequest
+from django.utils.translation import gettext_lazy as _
+
+import requests
+from djangorestframework_camel_case.render import CamelCaseJSONRenderer
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import OpenApiParameter, extend_schema
+from rest_framework import status
+from rest_framework.generics import GenericAPIView
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from openforms.api.serializers import ExceptionSerializer, ValidationErrorSerializer
+from openforms.submissions.api.permissions import AnyActiveSubmissionPermission
+
+from .serializers import MapSearchSerializer
+
+logger = logging.getLogger(__name__)
+
+
+@extend_schema(
+    summary=_("List BAG address suggestions."),
+    parameters=[
+        OpenApiParameter(
+            "q",
+            OpenApiTypes.STR,
+            OpenApiParameter.QUERY,
+            description=_("The search query we send to the pdok locatieserver api."),
+            required=True,
+        )
+    ],
+    responses={
+        200: MapSearchSerializer,
+        status.HTTP_400_BAD_REQUEST: ValidationErrorSerializer,
+        status.HTTP_401_UNAUTHORIZED: ExceptionSerializer,
+    },
+)
+class MapSearchView(GenericAPIView):
+    serializer_class = MapSearchSerializer
+    permission_classes = [AnyActiveSubmissionPermission]
+    renderer_classes = [CamelCaseJSONRenderer]
+
+    def get(self, request: Request, *args, **kwargs):
+        query = request.GET.get("q")
+        if not query:
+            return HttpResponseBadRequest(_("Missing query parameter 'q'"))
+
+        url = f"{settings.PDOK_LOCATIE_SERVER_URL}free"
+        data = {"q": query}
+
+        try:
+            bag_data = requests.get(url, params=data)
+        except requests.exceptions.RequestException as e:
+            logger.exception(f"couldn't retrieve pdok locatieserver data: {e}")
```

Comment:
doesn't this also cover connection errors and SSL errors etc.? which would lead to misconfiguration, so you'd want to properly see the necessary information.

You also don't want to interpolate information in error messages like that, especially not with f-strings. the `logger.execption` call already includes the exception information which we can see in Sentry.

Reply:
How can we differentiate between misconfiguration and connection errors? Because I don't see how we can tell a service that is down apart from a service that doesn't exists.  

Is the code that we have now sufficient? Because we are doing the same thing now as in the haalcentraal/client.py